pkg/operator/admission/webhook.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/ceph/csi"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"os"
+)
+
+const (
+	appName            = "rook-ceph-admission-controller"
```

Comment:
The appName and service account names should also be parameters to the StartServer() so we can remove ceph from this package. (Perhaps there should be a struct for the settings required). Other storage providers should also be able to use this same admission controller package to start their controllers.

Reply:
I'm thinking we need to define a struct that has all the provider-specific parameters. For example:
```
type ProviderConfig struct {
  // add fields for the app name, handler function (instead of passing it to `StartServer()`)
  // also add fields for other context instead of using package vars, such as the `context` var that is defined in server.go
}
```
Then we can finish this PR with everything needed for other storage providers to create an admission controller.